### PR TITLE
Allow placing all solar thermal collectors on a PlantEquipmentList

### LINKED ISSUE #3761: IntegratedCollectorStorage object being incorrectly added to the PlantEquipmentList
When trying to add an integrated collector storage solar collector, the plant equipment list is being generated with the collector, but it doesn't seem to be pulling in the equipment name correctly. Flat plate collectors seem to be working fine, this issue is specific to ICS collectors. Attached is an OSM created by our measures with the ICS that doesn't run in EnergyPlus; manually adding the collector name does allow the simulation to run.

[run.zip](https://github.com/NREL/OpenStudio/files/3844036/run.zip)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #3761
 - Allow placing all solar collectors on a `PlantEquipmentList` (`SolarCollectorFlatPlatePhotovoltaicThermal` and `SolarCollectorIntegralCollectorStorage`) by adding the proper `\reference validPlantEquipmentNames` to the `ProposedEnergy+.idd`
 - Add a unit test for it too.
 - Extend OpenStudio-resources regression tests: https://github.com/NREL/OpenStudio-resources/pull/85

### Pull Request Author

 - [x] Model API Changes / Additions: N/A
 - [x] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate): N/A
 - [x] Model API methods are tested (in `src/model/test`): N/A
 - [x] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`): **Yes, added some**
 - [x] If a new object or method, added a test in NREL/OpenStudio-resources: https://github.com/NREL/OpenStudio-resources/pull/85
 - [x] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`): N/A (no model API changes)
 - [x] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`): N/A
 - [x] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.: N/A
 - [x] All new and existing tests passes: **Yes**
 - [x] If methods have been deprecated, update rest of code to use the new methods: N/A

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: openstudiocore/resources/energyplus/ProposedEnergy+.idd [hunk 1]
```diff
@@ -40252,10 +40252,10 @@ SolarCollector:FlatPlate:Water,
   A1 , \field Name
        \required-field
        \type alpha
-       \reference-class-name validPlantEquipmentTypes ! remove?
-       \reference validPlantEquipmentNames ! remove?
        \reference-class-name validBranchEquipmentTypes
        \reference validBranchEquipmentNames
+       \reference-class-name validPlantEquipmentTypes ! Need to be able to place it on PlantEquipmentList
+       \reference validPlantEquipmentNames ! Need to be able to place it on PlantEquipmentList
```
Comment (jmarrec) [inline openstudiocore/resources/energyplus/ProposedEnergy+.idd]: No change really here, they were already there, just trying to be consistent

File: openstudiocore/resources/energyplus/ProposedEnergy+.idd [hunk 2]
```diff
@@ -40284,6 +40284,8 @@ SolarCollector:FlatPlate:PhotovoltaicThermal,
        \reference validOASysEquipmentNames
        \reference-class-name validBranchEquipmentTypes
        \reference validBranchEquipmentNames
+       \reference-class-name validPlantEquipmentTypes ! Need to be able to place it on PlantEquipmentList
+       \reference validPlantEquipmentNames ! Need to be able to place it on PlantEquipmentList
```
Comment (jmarrec) [inline openstudiocore/resources/energyplus/ProposedEnergy+.idd]: Fix 1/2

File: openstudiocore/resources/energyplus/ProposedEnergy+.idd [hunk 3]
```diff
@@ -40351,6 +40353,8 @@ SolarCollector:IntegralCollectorStorage,
        \type alpha
        \reference-class-name validBranchEquipmentTypes
        \reference validBranchEquipmentNames
+       \reference-class-name validPlantEquipmentTypes ! Need to be able to place it on PlantEquipmentList
+       \reference validPlantEquipmentNames ! Need to be able to place it on PlantEquipmentList
```
Comment (jmarrec) [inline openstudiocore/resources/energyplus/ProposedEnergy+.idd]: Fix 2/2

File: openstudiocore/src/energyplus/Test/PlantEquipmentOperationSchemes_GTest.cpp [hunk 4]
```diff
@@ -408,3 +409,67 @@ TEST_F(EnergyPlusFixture, ForwardTranslator_PlantEquipmentOperationSchemes_Water
...
+    EXPECT_EQ("SolarCollector:FlatPlate:Water", idf_eg.getString(PlantEquipmentListExtensibleFields::EquipmentObjectType).get());
+    EXPECT_EQ(collector_flatplate_water.name().get(), idf_eg.getString(PlantEquipmentListExtensibleFields::EquipmentName).get());
+
+    idf_eg = idf_peq_list.extensibleGroups()[1];
+    EXPECT_EQ("SolarCollector:FlatPlate:PhotovoltaicThermal", idf_eg.getString(PlantEquipmentListExtensibleFields::EquipmentObjectType).get());
+    EXPECT_EQ(collector_flatplate_pvt.name().get(), idf_eg.getString(PlantEquipmentListExtensibleFields::EquipmentName).get());
+
+    idf_eg = idf_peq_list.extensibleGroups()[2];
+    EXPECT_EQ("SolarCollector:IntegralCollectorStorage", idf_eg.getString(PlantEquipmentListExtensibleFields::EquipmentObjectType).get());
+    EXPECT_EQ(collector_integralcollectorstorage.name().get(), idf_eg.getString(PlantEquipmentListExtensibleFields::EquipmentName).get());
```
Comment (jmarrec) [inline openstudiocore/src/energyplus/Test/PlantEquipmentOperationSchemes_GTest.cpp]: Before fix:

```
PlantEquipmentList,
  Plant Loop 1 Heating Equipment List,    !- Name
  SolarCollector:FlatPlate:Water,         !- Equipment Object Type 1
  Solar Collector Flat Plate Water 1,     !- Equipment Name 1
  SolarCollector:FlatPlate:PhotovoltaicThermal, !- Equipment Object Type 2
  ,                                       !- Equipment Name 2
  SolarCollector:IntegralCollectorStorage, !- Equipment Object Type 3
  ;                                       !- Equipment Name 3
```

After:

```
PlantEquipmentList,
  Plant Loop 1 Heating Equipment List,    !- Name
  SolarCollector:FlatPlate:Water,         !- Equipment Object Type 1
  Solar Collector Flat Plate Water 1,     !- Equipment Name 1
  SolarCollector:FlatPlate:PhotovoltaicThermal, !- Equipment Object Type 2
  Solar Collector Flat Plate Photovoltaic Thermal 1,     !- Equipment Name 2
  SolarCollector:IntegralCollectorStorage, !- Equipment Object Type 3
  Solar Collector Integral Collector Storage 1;    !- Equipment Name 3
```

**Comment by shorowit** [comment]
Thanks!

**Pull request merged**